components/suggestion/command_provider/applet_command_parser.ts

```diff
@@ -0,0 +1,589 @@
+
+    let tokens = getTokens(pretext);
+
+    tokens = tokens.slice(binding.fullPretext?.split(' ').length);
+    const lastToken = tokens.pop();
+
+    const position = tokens.length;
+
+    const field = positionalDefinitions[position];
+    if (field) {
+        return getSuggestionsForField(field, lastToken.value, pretext);
+    }
+
+    const previousToken = tokens[position-1];
+    if (previousToken && previousToken.type === 'flag') {
+        const field = form.fields.find((arg) => arg.name === previousToken.name);
+        if (field) {
+            return getSuggestionsForField(field, lastToken.value, pretext);
+        }
+        return [];
+    }
+
+    // '', '-', or '--' (show named args)
+    if ('--'.startsWith(lastToken.value.trim())) {
+        const available = form.fields.filter((arg) => {
+            if (arg.positional) {
+                return false;
+            }
+            if (tokens.find((t) => t.type === 'flag' && t.name === arg.name)) {
+                return false;
+            }
+            if (arg.name.startsWith(lastToken.name)) {
+                return true;
+            }
+
+            return false;
+        });
+        return getSuggestionsFromFlags(available);
+    }
+
+    return [{suggestion: 'Default Argument Suggestion'}];
+};
+
+export const getSuggestionsForField = async (field: AutocompleteElement, userInput: string, pretext: string): Promise<SuggestionChoice[]> => {
+    switch (field.type) {
+        case AppletFieldTypes.BOOL:
+            return getBooleanSuggestions(userInput);
+        case AppletFieldTypes.DYNAMIC_SELECT:
+            return getDynamicSelectSuggestions(field as AutocompleteDynamicSelect, userInput, pretext)
+        case AppletFieldTypes.STATIC_SELECT:
+            return getStaticSelectSuggestions(field as AutocompleteStaticSelect, userInput)
+    }
+
+    return [{
+        complete: '',
+        suggestion: userInput,
+        description: field.description,
+        hint: field.hint,
+    }];
+}
+
+export const getStaticSelectSuggestions = (field: AutocompleteStaticSelect, userInput: string): SuggestionChoice[] => {
+    const opts = field.options.filter((opt) => opt.label.toLowerCase().startsWith(userInput.toLowerCase()));
+    return opts.map((opt) => ({
+        complete: opt.label,
+        suggestion: opt.label,
+        // hint: opt.label,
+        description: '',
+    }));
+}
+
+export const getDynamicSelectSuggestions = async (field: AppletField, userInput: string, cmdStr: string): Promise<SuggestionChoice[]> => {
+    const values = getForm(cmdStr, COMMANDS);
+
+    const cmd = getMatchedCommand(cmdStr, COMMANDS);
+
+    const payload: AppletCall = {
+        url: field.source_url || '',
+        context: {
+            ...ARGS,
+            app_id: cmd.app_id,
+        },
+        values,
+        raw_command: cmdStr,
+    };
+
+    const res = await doAppletCall<{}, AppletSelectOption[]>(payload);
```

Comment:
This will become doAppletSelectLookup or something like that, eventually? Or is the intent to leverage `Call` for lookups as well?

Reply:
We are leveraging Call in this PR. Seems fitting since the `Context` is available